### PR TITLE
satellite maintain metadata

### PR DESCRIPTION
**Pull request by evgeni**
- **add test for Satellite Upgrade scenario confinement**
- **assert target_version**
- **move the logic which versions are supported into the downstream concern**

### CODE REVIEW

File: lib/foreman_maintain/concerns/downstream.rb [hunk 1]
```diff
@@ -46,8 +48,25 @@ def fm_pkg_and_cmd_name
...
+          ForemanMaintain.upgrade_in_progress == satellite_maintain_config['current_sat_version']
+      end
+
       private
 
+      def satellite_maintain_config
+        if File.exist?(SATELLITE_MAINTAIN_CONFIG)
+          YAML.load_file(SATELLITE_MAINTAIN_CONFIG)
+        else
+          {}
```
Comment (ehelms) [inline lib/foreman_maintain/concerns/downstream.rb:66]: This is a good sane default, but I wondering if we shouldn't throw a more in-yo-face error if the config file is missing as that indicates the system is likely in a bad state, something is missing and you can't proceed.

File: test/definitions/test_helper.rb [hunk 2]
```diff
@@ -209,6 +209,15 @@ def assume_service_missing(name, priority = 30)
       yield service if block_given?
     end
   end
+
+  def mock_satellite_maintain_config
+    config = {
+      'current_sat_version' => '6.16',
+      'previous_sat_version' => '6.15',
+    }
```
Comment (ehelms) [inline test/definitions/test_helper.rb:217]: My 2-cents, either spell out `satellite`, drop the `sat` or nest the data:

```
satellite: {
  current_version
  previous_version
}
```

**Comment by evgeni** [comment]
@ehelms updated! (and offended the :policeman:)

File: lib/foreman_maintain/concerns/downstream.rb [hunk 3]
```diff
@@ -46,8 +48,25 @@ def fm_pkg_and_cmd_name
...
+          ForemanMaintain.upgrade_in_progress == satellite_maintain_target_version
+      end
+
       private
 
+      def satellite_maintain_config
+        if File.exist?(SATELLITE_MAINTAIN_CONFIG)
+          YAML.load_file(SATELLITE_MAINTAIN_CONFIG)
+        else
+          raise "Couldn't load satellite-maintain configuration file #{SATELLITE_MAINTAIN_CONFIG}."
```
Comment (ehelms) [inline lib/foreman_maintain/concerns/downstream.rb:66]:
```suggestion
          raise "Could not load satellite-maintain configuration file #{SATELLITE_MAINTAIN_CONFIG}."
```

Contractions get messy

**Review by ehelms — APPROVED** [review]
Small nit

**Pull request merged**